src/main/java/org/broadinstitute/hellbender/tools/spark/ParallelCopyGCSDirectoryIntoHDFSSpark.java

```diff
@@ -0,0 +1,171 @@
+ * concat method is used to stitch together chunks into single files without re-copying them.
+ */
+@CommandLineProgramProperties(summary="Parallel copy a file or directory (non-recursive) from GCS into HDFS",
+        oneLineSummary="Parallel copy a file or directory from GCS into HDFS.",
+        programGroup = SparkProgramGroup.class)
+public class ParallelCopyGCSDirectoryIntoHDFSSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc = "input GCS file path (add trailing slash when specifying a directory)", fullName = "inputGCSPath")
+    private String inputGCSPath = null;
+
+
+    @Argument(doc = "output directory on HDFS", shortName = "outputHDFSDirectory",
+            fullName = "outputHDFSDirectory")
+    private String outputHDFSDirectory;
+
+
+    @Override
+    protected void runTool(final JavaSparkContext ctx) {
+
+        final AuthHolder auth = getAuthHolder();
+        final String inputGCSPathFinal = inputGCSPath;
+        final String outputDirectoryFinal = outputHDFSDirectory;
+
+        org.apache.hadoop.fs.Path file = new org.apache.hadoop.fs.Path(outputHDFSDirectory);
+
+        try(FileSystem fs = file.getFileSystem(new Configuration())){
+
+            fs.mkdirs(file);
+            final long chunkSize = Long.parseLong(fs.getConf().get("dfs.blocksize"));
+
+            final List<Path> gcsPaths;
+            final Path inputGCSNIOPath = IOUtils.getPath(inputGCSPathFinal);
+            if (Files.isDirectory(inputGCSNIOPath)) {
+                logger.warn("directory: " + inputGCSPathFinal);
+                gcsPaths = Files.list(inputGCSNIOPath).collect(Collectors.toList());
+            } else {
+                logger.warn("single file");
+                gcsPaths = Collections.singletonList(inputGCSNIOPath);
+            }
+
+            List<Tuple2<String, Integer>> chunkList = new ArrayList<>();
+            for (Path path : gcsPaths) {
+
+                if (Files.isDirectory(path)) {
+                    logger.info("skipping directory " + path);
+                    continue;
+                }
+
+                final long size = Files.size(path);
+                final long chunks = size / chunkSize + 1;
+                logger.info("processing path " + path + ", size = " + size + ", chunks = " + chunks);
+
+                for (int i = 0; i < chunks; i++) {
+                    chunkList.add(new Tuple2<>(path.toUri().toString(), i));
+                }
+            }
+
+            if (chunkList.size() == 0) {
+                logger.info("no files found to copy");
+                return;
+            }
+
+            final JavaPairRDD<String, Integer> chunkRDD = ctx.parallelizePairs(chunkList, chunkList.size());
+
+            final JavaPairRDD<String, Tuple2<Integer, String>> chunkMappingRDD =
+                    chunkRDD.mapToPair(p -> new Tuple2<>(p._1(), readChunkToHdfs(auth, p._1(), chunkSize, p._2(), outputDirectoryFinal)));
```

Comment:
I'll add something, although you also get the Spark stage progress bar for chunk progress.